build_library/grub.cfg

```diff
@@ -47,28 +50,34 @@ function load_coreos {
         set suf=""
     fi
 
-    echo "Loading ($root)/boot/vmlinuz"
-    linux$suf /boot/vmlinuz $linux_console $linux_root "$@" $linux_append
+    search --no-floppy --set root --part-label EFI-SYSTEM
+    set kernel="/coreos/vmlinuz-$image"
 
-    if [ -f /boot/initrd ]; then
-        echo "Loading ($root)/boot/initrd"
-        initrd$suf /boot/initrd
-    fi
+    echo "Loading ($root)$kernel"
+    linux$suf $kernel $linux_console $linux_root "$1" $linux_append
 
     echo "Booting CoreOS!"
 }
 
 menuentry "CoreOS default" --id=coreos {
     gptprio.next -d root -u usr_uuid
+    if [ "$root" = "hd0,gpt3" ]; then
```

Comment:
Assuming the name 'hd0' isn't going to be reliable, also I'd like to avoid making assumptions about our partition numbers even if the assumption is safe today. Perhaps we name kernel based on partition UUID or label so there isn't a translation from gptprio.next or if you do translate the UUIDs are currently static:

USR-A: 7130c94a-213a-4e5a-8e26-6cce9662f132
USR-B: e03dd35c-7c2d-4a47-b3fe-27f15780a57c

These are defined in scripts/build_library/disk_layout.json

Using the label USR-A or USR-B would look prettier but that requires adding a new option to the gptprio.next command in grub/grub-core/commands/gptprio.c